drivers/serial/uart_nrfx_uart.c

```diff
@@ -0,0 +1,419 @@
+static inline void uart_nrfx_irq_tx_disable(struct device *dev)
+{
+	nrf_uart_int_disable(NRF_UART0, NRF_UART_INT_MASK_TXDRDY);
+}
+
+/** Interrupt driven transfer ready function */
+static inline int uart_nrfx_irq_tx_ready(struct device *dev)
+{
+	return nrf_uart_event_check(NRF_UART0, NRF_UART_EVENT_TXDRDY);
+}
+
+/** Interrupt driven receiver enabling function */
+static inline void uart_nrfx_irq_rx_enable(struct device *dev)
+{
+	nrf_uart_int_enable(NRF_UART0, NRF_UART_INT_MASK_RXDRDY);
+}
+
+/** Interrupt driven receiver disabling function */
+static inline void uart_nrfx_irq_rx_disable(struct device *dev)
+{
+	nrf_uart_int_disable(NRF_UART0, NRF_UART_INT_MASK_RXDRDY);
+}
+
+/** Interrupt driven transfer empty function */
+static inline int uart_nrfx_irq_tx_complete(struct device *dev)
+{
+	return !nrf_uart_event_check(NRF_UART0, NRF_UART_EVENT_TXDRDY);
+}
+
+/** Interrupt driven receiver ready function */
+static inline int uart_nrfx_irq_rx_ready(struct device *dev)
+{
+	return nrf_uart_event_check(NRF_UART0, NRF_UART_EVENT_RXDRDY);
+}
+
+/** Interrupt driven error enabling function */
+static inline void uart_nrfx_irq_err_enable(struct device *dev)
+{
+	nrf_uart_int_enable(NRF_UART0, NRF_UART_INT_MASK_ERROR);
+}
+
+/** Interrupt driven error disabling function */
+static inline void uart_nrfx_irq_err_disable(struct device *dev)
+{
+	nrf_uart_int_disable(NRF_UART0, NRF_UART_INT_MASK_ERROR);
+}
+
+
+
+/** Interrupt driven pending status function */
+static inline int uart_nrfx_irq_is_pending(struct device *dev)
+{
+	return ((nrf_uart_int_enable_check(NRF_UART0,
+					   NRF_UART_INT_MASK_TXDRDY) &&
+		uart_nrfx_irq_tx_ready(dev))
+		||
+		(nrf_uart_int_enable_check(NRF_UART0,
+					   NRF_UART_INT_MASK_RXDRDY) &&
+		uart_nrfx_irq_rx_ready(dev)));
+}
+
+/** Interrupt driven interrupt update function */
+static inline int uart_nrfx_irq_update(struct device *dev)
+{
+	return 1;
+}
+
+/** Set the callback function */
+static inline void uart_nrfx_irq_callback_set(struct device *dev,
+					      uart_irq_callback_t cb)
+{
+	(void)dev;
+	m_irq_callback = cb;
+}
+
+/**
+ * @brief Interrupt service routine.
+ *
+ * This simply calls the callback function, if one exists.
+ *
+ * @param arg Argument to ISR.
+ *
+ * @return N/A
+ */
+static void uart_nrfx_isr(void *arg)
+{
+	struct device *dev = arg;
+
+	if (m_irq_callback) {
+		m_irq_callback(dev);
+	}
+}
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+
+DEVICE_DECLARE(uart_nrfx_uart0);
+
+/**
+ * @brief Initialize UART channel
+ *
+ * This routine is called to reset the chip in a quiescent state.
+ * It is assumed that this function is called only once per UART.
+ *
+ * @param dev UART device struct
+ *
+ * @return 0 on success
+ */
+static int uart_nrfx_init(struct device *dev)
+{
+	struct device *gpio_dev;
+	int err;
+
+	gpio_dev = device_get_binding(CONFIG_GPIO_NRF5_P0_DEV_NAME);
+
+	__ASSERT(gpio_dev,
+			 "UART init failed. Cannot find %s",
+			 CONFIG_GPIO_NRF5_P0_DEV_NAME);
```

Comment:
This indentation is a bit weird. One TAB too far?

Reply:
I used 4 character tab instead of 8, I will fix it.